Yoast/wordpress-seo
# Adjust paragraph length criteria for product pages

## Context


*

## Summary


This PR can be summarized in the following changelog entry:

* [yoastseo] Refactors the `paragraphTooLong` assessment into class and adds custom config for the assessment when used for product pages.

## Relevant technical choices:

*

## Test instructions

### Test instructions for the acceptance test before the PR gets merged
This PR can be acceptance tested by following these steps:

*


### Test instructions for QA when the code is in the RC


- Set the following config in `scoring/contentAssessor.js` and in `/scoring/cornerstone/contentAssessor.js` file for `ParagraphTooLong` assessment:
` { parameters: {
			recommendedLength: 70,
			maximumRecommendedLength: 100,
			} }` (don't commit anything):

- Build the free plugin
- Create a post and add some text
- Create a paragraph with less than 70 words and confirm that the bullet turns green and reads: `None of the paragraphs are too long. Great job!`
- Create a paragraph with a number of words between 70 and 100 and confirm that the bullet turns orange and reads: `1 of the paragraphs contains more than the recommended maximum of 70 words. Shorten your paragraphs!`
- Create a paragraph with more than 100 words and confirm that the bullet turns red and reads: `1 of the paragraphs contains more than the recommended maximum of 70 words. Shorten your paragraphs!`
- Enable a cornerstone article
- Confirm that you get the same results as for a normal product page



QA can test this PR by following these steps:

*

## Impact check

This PR affects the following parts of the plugin, which may require extra testing:

*

## UI changes


## Documentation


## Quality assurance


Fixes #
